wardenclyffe/main/models.py

```diff
@@ -676,25 +676,27 @@ def process(self):
         self.status = "in progress"
         self.save()
         f = self.get_task()
-        error_message = ""
         try:
             (success, message) = f(self)
             self.status = success
             if self.status == "failed" or message != "":
                 self.log(info=message)
-                error_message = message
+                self.fail(message)
         except Exception, e:
-            self.status = "failed"
             self.log(info=str(e))
-            error_message = str(e)
+            # re-raise so Celery's retry logic can deal with it
+            raise
 
         self.save()
```

Comment:
Would it make sense to put this save/post_process test into the try block? Might make the code a bit clearer here.